gradle-plugins/build.gradle.kts

```diff
@@ -18,5 +18,10 @@ gradlePlugin {
 }
 
 dependencies {
-    implementation(libs.kotlinx.serialization.json)
+    // `kotlin-dsl` applied here uses embedded Kotlin version 1.9.23 for gradle wrapper 8.9
+    // https://docs.gradle.org/current/userguide/compatibility.html#kotlin
+    // This is needed as newer versions of kotlinx-serialization has been compiled using Kotlin 2.0+
+    // https://github.com/Kotlin/kotlinx.serialization/releases
+    // Sample modules do not use kotlin-dsl, instead uses latest kotlin serialization from libs versions.
+    implementation("org.jetbrains.kotlinx:kotlinx-serialization-json:1.6.3")
```

Comment:
@puneet-pdx Added a comment which hope helps explains the need for the revert change to keep in sync with toolkit. Ready for your review!